### PR TITLE
add an option to the sync command to 're-sync' images that already exist

### PR DESCRIPTION
**Pull request by alex-phillips**
Since XMP sidecars are now allowed, images with the same checksum will not get updated metadata if the sidecar changes but the checksum remains the same. This adds an optional flag to the `lychee:sync` command that forces existing images with no change in checksum to get updated metadata from XMP files (if they exist). It only updates the image if the metadata read in differs from the metadata that the image aleady has.

### CODE REVIEW

File: app/ModelFunctions/PhotoFunctions.php [hunk 1]
```diff
@@ -787,4 +794,37 @@ public function getValidExtensions(): array
...
+	 *
+	 * @return void
+	 */
+	private function getFileMetadata($file, $path, $kind, $mimeType, $extension)
+	{
+		if ($kind == 'raw') {
+			$info = $this->metadataExtractor->bare();
+			$this->metadataExtractor->size($info, $path);
+			$this->metadataExtractor->validate($info);
+			$info['type'] = 'raw';
```
Comment (tmp-hallenser) [inline app/ModelFunctions/PhotoFunctions.php]: #574 changes support of raw files, i.e. metadata is also extracted for raw files if possible. See https://github.com/LycheeOrg/Lychee/blob/32b460e93e56b50e0d49c25e4cf7a0e12968e282/app/ModelFunctions/PhotoFunctions.php#L286-L289

**Comment by alex-phillips** [comment]
It looks like this may have been an issue downloading EXIFtool and not the code changes?

File: app/ModelFunctions/PhotoFunctions.php (hunk 1 above)
Comment (alex-phillips) [reply app/ModelFunctions/PhotoFunctions.php]: @tmp-hallenser I adjusted my code to match the https://github.com/LycheeOrg/Lychee/pull/574

**Comment by ildyria** [comment]
> It looks like this may have been an issue downloading EXIFtool and not the code changes?

Yes. I'm currently doing a PR to fix that one.

**Pull request merged**